### PR TITLE
Update license and enhance collect_licenses script

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

This PR updates the LICENSE and NOTICE files for both source release and binary release, and enhances the `collect_licenses.sh` script to support 1)collect `NOTICE*`(previous only `NOTICE`) files, 2) remove redundant or duplicated text from final NOTICE

This PR also fix the LICENSE and NOTICE files for shaded modules

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -46,7 +46,10 @@ ENV CI ${CI}
 ADD . /workspace/kyuubi
 WORKDIR /workspace/kyuubi
 
-RUN ./build/dist ${MVN_ARG} && \
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive \
+    apt-get install -y python3 && \
+    ./build/dist ${MVN_ARG} && \
```
Comment (ulysses-you) [inline build/Dockerfile:52]: why we do the change ?

Comment (pan3793) [reply build/Dockerfile:52]: Add a new `build/release/append_notice.py` to reduce the redundant NOTICE

File: LICENSE [hunk 2]
```diff
@@ -199,17 +199,3 @@
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
-
```
Comment (yaooqinn) [inline LICENSE:202]: For source release, we now bundled

src/main/resources/org/apache/kyuubi/ui/swagger/*
src/main/resources/org/apache/kyuubi/ui/static/assets/fonts/*
src/main/resources/org/apache/kyuubi/ui/static/semantic.min.css
src/main/resources/org/apache/kyuubi/ui/static/semantic.min.js
src/main/resources/org/apache/kyuubi/ui/static/icon.min.css
src/main/resources/org/apache/kyuubi/ui/static/jquery-3.6.0.min.js

File: LICENSE-binary [hunk 3]
```diff
@@ -270,27 +288,32 @@ org.apache.zookeeper:zookeeper
...
+org.glassfish.hk2:hk2-api
+org.glassfish.hk2:hk2-locator
+org.glassfish.hk2:hk2-utils
+org.glassfish.hk2.external:aopalliance-repackaged
+org.glassfish.hk2.external:jakarta.inject
+org.glassfish.hk2:osgi-resource-locator
 
 MIT License
 -----------
-org.slf4j:slf4j-api
```
Comment (yaooqinn) [inline LICENSE-binary:292]: For binary release, we now bundled

src/main/resources/org/apache/kyuubi/ui/swagger/*
src/main/resources/org/apache/kyuubi/ui/static/assets/fonts/*
src/main/resources/org/apache/kyuubi/ui/static/semantic.min.css
src/main/resources/org/apache/kyuubi/ui/static/semantic.min.js
src/main/resources/org/apache/kyuubi/ui/static/icon.min.css
src/main/resources/org/apache/kyuubi/ui/static/jquery-3.6.0.min.js

**Comment by yaooqinn** [comment]
we need a folder `licenses` for source releases to deploy thrid-party licenses as we have licenses-binary for binary distribution

File: LICENSE (hunk 2 above)
Comment (pan3793) [reply LICENSE:202]: This is not a part of 1.5

**Comment by yaooqinn** [comment]
licenses-binary/LICENSE-CC0.txt seems not needed?

File: LICENSE (hunk 2 above)
Comment (yaooqinn) [reply LICENSE:202]: Let's fix the master first, then subtracting for 1.5

Comment (pan3793) [reply LICENSE:202]: where are `src/main/resources/org/apache/kyuubi/ui/static/assets/fonts/*` comes from? and what's the license of them?

Comment (yaooqinn) [reply LICENSE:202]: see icon.min.css's header

**Comment by pan3793** [comment]
Updated based on comments, please take a look again @yaooqinn

File: kyuubi-server/src/main/resources/org/apache/kyuubi/ui/swagger/oauth2-redirect.html [hunk 4]
```diff
@@ -1,3 +1,20 @@
+<!--
```
Comment (yaooqinn) [inline kyuubi-server/src/main/resources/org/apache/kyuubi/ui/swagger/oauth2-redirect.html:1]: this is not an apache file

**Comment by yaooqinn** [comment]
will the newly added `licenses` fold be deployed with source releases correctly?

File: kyuubi-server/src/main/resources/org/apache/kyuubi/ui/swagger/oauth2-redirect.html (hunk 4 above)
Comment (pan3793) [reply kyuubi-server/src/main/resources/org/apache/kyuubi/ui/swagger/oauth2-redirect.html:1]: The original file does not have license header, we can add our license header to the original file

**Comment by pan3793** [comment]
> will the newly added `licenses` fold be deployed with source releases correctly?

Yes, it's controlled by `.gitattributes`.

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
The failure caused by Trino container launch error, not a big deal since we not deliver the Trino support. Thanks all for helping review, I'm going to merge it and open a PR for branch-1.5

**Pull request closed without merging**